pkg/tasks/thanos_ruler.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2018 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tasks
+
+import (
+	"encoding/json"
+
+	"github.com/openshift/cluster-monitoring-operator/pkg/client"
+	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ThanosRulerTask struct {
+	client  *client.Client
+	factory *manifests.Factory
+}
+
+func NewThanosRulerTask(client *client.Client, factory *manifests.Factory) *ThanosRulerTask {
+	return &ThanosRulerTask{
+		client:  client,
+		factory: factory,
+	}
+}
+
+func (t *ThanosRulerTask) Run() error {
+	svc, err := t.factory.ThanosRulerService()
```

Comment:
as thanos ruler is part of the user workload monitoring stack, it should only be deployed if that is active.

Reply:
Moved everything over to user workload monitoring